### DISCUSSION TITLE
[월드 1.1.7] EventFunction중 OnMouseXXX 호출 안되는 현상

### DISCUSSION THREAD

**Discussion by smilejsu82** [naverz/zepeto-studio-kor#619] [5. Q&A]
월드 버전 1.1.7 입니다 

이벤트 함수중 다음 함수 들이 호출이 안되는것 같습니다 
확인좀 부탁합니다 :)

`OnMouseDown() {
        console.log('OnMouseDown.');
    }
    OnMouseDrag() {
        console.log('OnMouseDrag.');
    }  
    OnMouseUp() {
        console.log('OnMouseUp.');
    }  
    OnMouseEnter() {
        console.log('OnMouseEnter.');
    } 
    OnMouseExit() {
        console.log('OnMouseExit.');
    }  
    OnMouseOver() {
        console.log('OnMouseOver.');
    }  
    OnMouseUpAsButton() {
        console.log('OnMouseUpAsButton.');
    }   `

![image](https://user-images.githubusercontent.com/50188795/163404151-d85ec204-8e5c-4273-9b7c-4bf25e0ff9ad.png)
![image](https://user-images.githubusercontent.com/50188795/163404193-c7a57097-1256-4740-9dd2-8ff260f08345.png)

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. 동일한 1.1.7에서 동작이 확인되었는데 다시 한번 테스트 해보시겠어요? 

<img width="1125" alt="스크린샷 2022-04-15 오전 10 48 32" src="https://user-images.githubusercontent.com/100118650/163505056-4e7fd434-0bea-4a51-9445-e2d15aff3ebf.png">

**Reply by smilejsu82** [reply]
월드 1.1.7로 테스트 해보았습니다 
기본 제페토 템플릿 프로젝트 샘플 씬에 있는 SampleScript에 추가 했습니다 
![image](https://user-images.githubusercontent.com/50188795/163526814-ff779228-f5cd-4057-8fac-6911a7ed636a.png)
![image](https://user-images.githubusercontent.com/50188795/163526828-83d0a5d8-2f96-44c3-a402-b33de2958429.png)
![image](https://user-images.githubusercontent.com/50188795/163526837-fd4cff25-3949-4a07-88b0-c206fe18111d.png)
![image](https://user-images.githubusercontent.com/50188795/163526877-465ac6b3-dca3-4b86-9413-8cbf432938a4.png)
![image](https://user-images.githubusercontent.com/50188795/163526861-5974060a-eb49-4df1-94a4-fad75411f202.png)

동작 하지 않습니다...

**Reply by smilejsu82** [reply]
이것은 제페토 프로젝트가 아닌 일반 프로젝트 3D템플릿 입니다 
잘 동작 합니다 

![image](https://user-images.githubusercontent.com/50188795/163527177-860c2c6d-a650-4e2c-aece-57228d3ead1a.png)

**Reply by smilejsu82** [reply]
https://docs.unity3d.com/kr/530/ScriptReference/MonoBehaviour.OnMouseDown.html

![image](https://user-images.githubusercontent.com/50188795/163529775-237e240b-0e8f-4e0a-8c28-28bdd3535a4e.png)

혹시 제가 빠트린게 있을까요?

**Reply by smilejsu82** [reply]
올려주신 결과물 스샷에서 

![image](https://user-images.githubusercontent.com/50188795/163530063-95e9e371-2d6b-4c88-b3f7-a89f479bb3b7.png)
![image](https://user-images.githubusercontent.com/50188795/163530178-e01b29d7-347d-46ac-8bc6-f2467d2e9fb0.png)


이거 스펠링이 좀 이상한데요?

**Reply by Zepeto-TechEvangelist** [reply]
스펠링은 그냥 클래스명 오타입니다.🥲 
아래와 같이 스크립트 작성해서 큐브에 연결했습니다. 

import { ZepetoScriptBehaviour } from 'ZEPETO.Script'

export default class MouseEvent extends ZepetoScriptBehaviour {
    
    OnMouseDown() {
        console.log('OnMouseDown.');
    }
    OnMouseDrag() {
        console.log('OnMouseDrag.');
    }
    OnMouseUp() {
        console.log('OnMouseUp.');
    }
    OnMouseEnter() {
        console.log('OnMouseEnter.');
    }
    OnMouseExit() {
        console.log('OnMouseExit.');
    }
    OnMouseOver() {
        console.log('OnMouseOver.');
    }
    OnMouseUpAsButton() {
        console.log('OnMouseUpAsButton.');
    }

}

**Reply by smilejsu82** [reply]
![image](https://user-images.githubusercontent.com/50188795/163533205-198a48fc-75ea-46cf-b07f-b277cb8cce23.png)
![image](https://user-images.githubusercontent.com/50188795/163533272-e739938a-3230-40e5-8ca0-c12bbe2179b7.png)
![image](https://user-images.githubusercontent.com/50188795/163533365-9730531f-3306-4be2-89dc-d746cbce31ed.png)

저도 똑같이 해서 방금 해봤는데요 
동작을 안하는데요ㅠ 
흠.. 전 왜 안되는것일까요

**Reply by Zepeto-TechEvangelist** [reply]
최신 멀티플레이 샘플 위에서 한번 해보시겠어요?

**Reply by smilejsu82** [reply]
> 최신 멀티플레이 샘플 위에서 한번 해보시겠어요? 
> 
> 

최신 멀티플레이는 요고를 말씀하시는 거죠? 
https://github.com/naverz/zepeto-multiplay-example

**Comment by smilejsu82** [comment]
와우...
여기서는 동작 하네요...?!
왜죠?!  ~ ,. ~ 

![Screen Shot 2022-04-15 at 오후 5 26](https://user-images.githubusercontent.com/50188795/163546274-9dd79bcc-d4c3-4807-905a-137af4d26e38.png)
![image](https://user-images.githubusercontent.com/50188795/163546375-1f06c333-e739-4483-9829-57444efa6cb7.png)

**Reply by smilejsu82** [reply]
감사합니다 
덕분에 해결 했습니다 👍 

![image](https://user-images.githubusercontent.com/50188795/163549401-bae56fb6-4f8b-4092-be49-2bad3e1db7e0.png)
![image](https://user-images.githubusercontent.com/50188795/163550136-a41d492f-c9e6-4602-a8cd-35eb9f539bc0.png)
![image](https://user-images.githubusercontent.com/50188795/163550158-01d96e08-5a7b-4038-aba9-6fad0a704be0.png)
